File: Leetcode/JS/Leetcode1576-替换所有的问好.js
var modifyString = function(s) {
    const rec = [...s];
    const n = s.length;
    for(let i = 0;i < n;i++){
        if(rec[i] == '?'){
            for(let j = 0;j < 3;j++){
                if ((i > 0 && rec[i - 1] === String.fromCharCode('a'.charCodeAt() + j)) || (i < n - 1 && rec[i + 1] === String.fromCharCode('a'.charCodeAt() + j))) {
                    continue;
                }
                rec[i] = String.fromCharCode('a'.charCodeAt() + j);
                break;
            }
        }
    }
    return rec.join('');
};

File: Leetcode/PY/Leetcode1576-替换所有的问好.py
class Solution:
    def modifyString(self, s: str) -> str:
        rec,n = list(s),len(s)
        for i in range(n):
            if rec[i] == '?':
                for t in 'abc':
                    if (i > 0 and rec[i - 1] == t) or (i < n - 1 and rec[i + 1] == t): continue
                    rec[i] = t
                    break
        return ''.join(rec)